src/core/pyproxy.c

```diff
@@ -1172,8 +1310,47 @@ EM_JS_NUM(int, pyproxy_init_js, (), {
   };
 
   Module.PyProxyCallableMethods = { prototype : Function.prototype };
-  Module.PyProxyBufferMethods = {};
+  Module.PyProxyBufferMethods = {
+    getRawBuffer : function(type = "u8"){
+      let ArrayType = type_to_array_map.get(type);
+      if(ArrayType === undefined){
+        throw new Error(`Unknown type ${type}`);
+      }
+      let this_ptr = _getPtr(this);
+      let buffer_struct_ptr = __pyproxy_memoryview_get_buffer(this_ptr);
+      if(buffer_struct_ptr === 0){
+        throw new Error("Failed");
+      }
 
+      // This has to match the order of the fields in buffer_struct
+      let cur_ptr = buffer_struct_ptr/4;
+      let start = HEAP32[cur_ptr++];
+      let smallest = HEAP32[cur_ptr++];
+      let largest = HEAP32[cur_ptr++];
+      let shape = Module.hiwire.pop_value(HEAP32[cur_ptr++]);
+      let strides = Module.hiwire.pop_value(HEAP32[cur_ptr++]);
+      let view_ptr = HEAP32[cur_ptr++];
+      _PyMem_Free(buffer_struct_ptr);
+
+
+      let alignment = parseInt(type.slice(1))/8;
+      if(start % alignment !== 0 || smallest % alignment !== 0 || largest % alignment !== 0){
+        _PyBuffer_Release(view_ptr);
+        _PyMem_Free(view_ptr);
+        throw new Error(`Buffer does not have valid alignment for type ${type}`);
+      }
+
+      let length = (largest - smallest) / alignment;
+      let offset = (start - smallest) / alignment;
+
+      let buffer = new ArrayType(HEAP8.buffer, smallest, length);
```

Comment:
I find these variable names a bit confusing. For instance in the Float32Array signature, `byteOffset` is significantly more explicit that `smallest`. And then in an ndarray what `length` could correspond to (size of the buffer, etc) is also unclear.

Reply:
I renamed these `minByteOffset`, `maxByteOffset`, `numBytes` and `numEntries`. Is it clear enough now?